courier/courier-impl/impl/src/java/org/sakaiproject/courier/impl/BasicCourierService.java

```diff
@@ -324,125 +272,68 @@ public boolean hasDeliveries(String address)
-		protected boolean m_maintenanceCheckerStop = false;
-
-		/**
-		 * Construct.
-		 */
-		public Maintenance()
-		{
-			M_log.info("Maintenance()");
-		}
-
-		/**
-		 * Start the maintenance thread.
-		 */
-		public void start()
-		{
-			if (m_maintenanceChecker != null) return;
-
-			m_maintenanceChecker = new Thread("Sakai.BasicCourierService.Maintenance");
-			m_maintenanceCheckerStop = false;
-			m_maintenanceChecker.setDaemon(true);
-			m_maintenanceChecker.start();
-		}
 
-		/**
-		 * Stop the maintenance thread.
-		 */
-		public void stop()
-		{
-			if (m_maintenanceChecker != null)
-			{
-				m_maintenanceCheckerStop = true;
-				m_maintenanceChecker.interrupt();
-				try
-				{
-					// wait for it to die
-					m_maintenanceChecker.join();
-				}
-				catch (InterruptedException ignore)
-				{
-				}
-				m_maintenanceChecker = null;
-			}
-		}
-
-		/**
-		 * Run the maintenance thread. Every m_checkEvery seconds, check for expired deliveries.
-		 */
-		public void run()
-		{
-			// since we might be running while the component manager is still being created and populated, such as at server
-			// startup, wait here for a complete component manager
-			ComponentManager.waitTillConfigured();
-			
-			M_log.debug("Maintenance thread running...");
-			try
-			{
-				long now = System.currentTimeMillis();
-				for (List<Delivery> deliveries : m_addresses.values()) {
-					for (Iterator<Delivery> iter = deliveries.iterator(); iter.hasNext();) {
-						Delivery delivery = iter.next();
-						if (delivery instanceof Expirable) {
-							Expirable expDelivery = (Expirable)delivery;
-							long created = expDelivery.getCreated();
-							int ttl = expDelivery.getTtl();
-							//Don't need to worry about it if ttl is not set.
-							if (ttl > 0) {
-								long sDiff = (now - created) / 1000;
-								// If the time since creation is larger than the ttl, remove it
-								if (sDiff > ttl) {
-									if (m_aggressiveCleanup) {
-										M_log.debug("Removing all (" + deliveries.size() + ") expired deliveries for address: " + delivery.getAddress());
-										clear(delivery.getAddress());
-										break;
-									}  
-									else {
-										M_log.debug("Removing a single expired delivery for address: " + delivery.getAddress());
-										iter.remove();
-									}
+	/**
+	 * Run the maintenance thread. Every m_checkEvery seconds, check for expired deliveries.
+	 */
+	@Override
+	public void run() {
+		Thread.currentThread().setName("Sakai.BasicCourierService.Maintenance");
+		// since we might be running while the component manager is still being created and populated, such as at server
+		// startup, wait here for a complete component manager
+		ComponentManager.waitTillConfigured();
+
+		log.debug("Maintenance thread start...");
+		try {
+			long now = System.currentTimeMillis();
+			for (List<Delivery> deliveries : m_addresses.values()) {
+				Iterator<Delivery> iter = deliveries.iterator();
```

Comment:
Originally was using for but there is a need to use the **iter** further down, so this is the right use case for using iterator. You will also note that Java 8's Collection.removIf() operates in the same way.